database/import_test.go

```diff
@@ -0,0 +1,324 @@
+				CumulatedWork: "4295032833",
+				PreviousBlock: "0000000000000000000000000000000000000000000000000000000000000000",
+			},
+			expectedErrorMessage: "",
+		},
+
+		{
+			testID: 2,
+			name:   "10 blocks beside the fork, should return valid block header",
+			data: testCaseData{
+				blockRecord: [][]string{
+					{"536870912", "0fb7334d7fc33e3284cc47e6682ea19d478716576fc02431977ee856d8a11a7a", "4179160797", "402791861", "1542300873"},
+					{"536870912", "334693eb277aa554d2c606041b97f03ba8689e8da373229375f5e5a1bbe5b5e3", "365081954", "402791587", "1542301036"},
+					{"536870912", "559b5d64cb554ba90f33b4047941dd4b9797600ed46b63c8d1cd71888b8c6565", "483339668", "402792728", "1542301708"},
+					{"536870912", "aa22501b3f7306aaeb7e222b3149becb874e11ff45d38c5e270b653cf3561ab4", "303622189", "402793092", "1542302315"},
+					{"536870912", "8049cf9f0e767e1b96998e0b9dd68da9242b7e0e108c83cc02edab38f5854aaa", "379456570", "402791605", "1542303567"},
+					{"536870912", "f093773ebea5c6a290d762d3d0b43eca7de94b5b25fd8a93c677f5fc63b3bf16", "1220874517", "402791706", "1542304321"},
+					{"536870912", "da2b9eb7e8a3619734a17b55c47bdd6fd855b0afa9c7e14e3a164a279e51bba9", "1301274612", "402792411", "1542305817"},
+					{"536870912", "b0ac8183e020907d399efb445e8fd6a90f611545c3f53ee9bab6130e1cf701a8", "522429575", "402792964", "1542306568"},
+					{"536870912", "dc3f21e9e6cfbe895e9fc39b39538ed771a4ba8b0be0cb414b89fbcfaeda691f", "889633737", "402795325", "1542307349"},
+					{"536870912", "17e0aefc0154e0a3cdc4a837c66d9c0e0f0e4a44a703fd6e654e8cbc62c0b28f", "4081063765", "402796026", "1542307497"},
+				},
+				previousBlockHash:  "000000000000000001f34f5eb45827af756e757498039f43ff6f7585c97f4d16",
+				blockHasher:        service.DefaultBlockHasher(),
+				cumulatedChainWork: "255327261802219463033558368",
+				rowIndex:           556761,
+				numberOfBlocks:     10,
+			},
+			expectedBlock: &dto.DbBlockHeader{
+				Height:        556770,
+				Hash:          "00000000000000000005569f09a80c66c8ebf514fdd1c03e803799c2420a4f5a",
+				Version:       536870912,
+				MerkleRoot:    "17e0aefc0154e0a3cdc4a837c66d9c0e0f0e4a44a703fd6e654e8cbc62c0b28f",
+				Timestamp:     timestampInLocalTime("2018-11-15 19:44:57+01:00"),
+				Bits:          402796026,
+				Nonce:         4081063765,
+				State:         "LONGEST_CHAIN",
+				Chainwork:     "2166624730970898396303",
+				CumulatedWork: "255349410425588691745638430",
+				PreviousBlock: "0000000000000000005013e7cc2889ada8b01f24dfc325d1398be82197fc623b",
+			},
+			expectedErrorMessage: "",
+		},
+		{
+			testID: 1,
+			name:   "newer block, should return valid block header",
+			data: testCaseData{
+				blockRecord: [][]string{
+					{"536870912", "e9446d4ebeb301aeb5a2f375ac062bf3581269d783362cf066f08bbe6040a885", "3035389718", "403300437", "1708954423"},
+				},
+				previousBlockHash:  "0000000000000000031817e0b646350cac1b8770d6cba60717e86185cadb15cc",
+				blockHasher:        service.DefaultBlockHasher(),
```

Comment:
It could be created just in test body, or even be a "const"-like variable

Reply:
Outdated because initialization of service.DefaultBlockHasher() has been moved to the calculateFields function, where it is used, so it does not need to be passed anymore. Tests updated